src/bp/ui-studio/src/web/views/OneFlow/diagram/VariablesEditor/index.tsx

```diff
@@ -0,0 +1,139 @@
+import { Button, FormGroup, InputGroup } from '@blueprintjs/core'
+import axios from 'axios'
+import { FlowVariable } from 'botpress/sdk'
+import { Dialog } from 'botpress/shared'
+import { FlowView } from 'common/typings'
+import React, { FC, Fragment, useEffect, useState } from 'react'
+import { connect } from 'react-redux'
+import { fetchFlows } from '~/actions'
+import { getCurrentFlow, RootReducer } from '~/reducers'
+
+import VariableModal from './Modal'
+
+interface OwnProps {}
+type StateProps = ReturnType<typeof mapStateToProps>
+type DispatchProps = typeof mapDispatchToProps
+
+type Props = StateProps & DispatchProps & OwnProps
+
+const VariablesEditor: FC<Props> = props => {
+  const [editingVariable, setEditingVariable] = useState<FlowVariable>(undefined)
+
+  const affectUpdateFlow = async (flow: FlowView) => {
+    await axios.post(`${window.BOT_API_PATH}/flow/${flow.location}`, { flow })
+    props.fetchFlows()
+  }
```

Comment:
Can you use the flow reducer instead? The action updateFlow can be used to update an existing flow.